### PR TITLE
Applayer plugin 5053 v3.15.1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/5053

Describe changes:
- get ready to use dynamic number of app-layer protos (also work with static constant) in all places
- preventive fix of macro with parenthesis cc @jufajardini 

Let me know if I should split this PR

#12163 first commits, just making alprotos number dynamic, and not yet adding plugins support

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.dcerpc_tcp | 40 | 42 | 105.0% | 

Pipeline 23707

**Comment by victorjulien** [comment]
replaced by #12256

**Comment by catenacyber** [comment]
Did you mean you do not want the PR split ?
If so, do you also want https://github.com/OISF/suricata/pull/12233 closed as taken in #12256 ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 24007

**Comment by victorjulien** [comment]
Confused, why is there a v15 open as well as a v16?

**Comment by catenacyber** [comment]
> Confused, why is there a v15 open as well as a v16?

TL;DR
This PR and https://github.com/OISF/suricata/pull/12307 are the current ones

v15.1 was a part of v16

I could have made a v16.1 which is v15.1 rebased, but not wasting resources

As QA showed something interesting cf https://github.com/OISF/suricata/pull/12307 I am closing https://github.com/OISF/suricata/pull/12256 

Let me know if
- you want me to rebase this one
- want a new complete rebased PR with this + https://github.com/OISF/suricata/pull/12307 + the other commits of https://github.com/OISF/suricata/pull/12256

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/12352
